algo/src/main/java/org/neo4j/graphalgo/impl/HugePageRank.java

```diff
@@ -262,7 +262,7 @@ private ComputeSteps createComputeSteps(
 
             tracker.add(sizeOfDoubleArray(partitionCount));
```

Comment:
And to the estimation methods as well: to `sharedUsage` in `memoryUsageFor`.

Reply:
If I think about it, the fact that we allocate the score array here is probably also an artifact from a previous implementation. We could move this to the constructor as well and maybe just use copyOf, since the initial values are the same.